### PR TITLE
Dedup policy implement between grpc and proxy plugin

### PR DESCRIPTION
**Pull request by zouyee**
Signed-off-by: zouyee <zounengren@cmss.chinamobile.com>

### 1. Why is this pull request needed and what does it do?
 `grpc and proxy` plugin  have the same policy implemented  which is worth to  be factored into a couple of common packages .

### 2. Which issues (if any) are related?
a part of https://github.com/coredns/coredns/issues/2695

### 3. Which documentation changes (if any) need to be made?
NONE

### 4. Does this introduce a backward incompatible change or deprecation?
NONE

### CODE REVIEW

**Review by miekg — CHANGES_REQUESTED** [review]
thanks.

dnsutil is probably the wrong package, I suggest pkg/policy and put everything in there

**Comment by zouyee** [comment]
@miekg PTAL

File: plugin/pkg/policy/policy.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	List(policy ...interface{}) []interface{}
+	String() string
+}
+
+// Random is a policy that implements random upstream selection.
+type Random struct{}
+
+var _ Policy = &Random{}
+
+// String return the name of policy Random
```
Comment (miekg) [inline plugin/pkg/policy/policy.go]:
```suggestion
// String returns the name of policy Random.
```

File: plugin/pkg/policy/policy.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+// RoundRobin is a policy that selects hosts based on round robin ordering.
+type RoundRobin struct {
+	robin uint32
+}
+
+var _ Policy = &RoundRobin{}
+
+// String return the name of policy RoundRobin
```
Comment (miekg) [inline plugin/pkg/policy/policy.go]:
```suggestion
// String returns the name of policy RoundRobin.
```

File: plugin/pkg/policy/policy.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+	return robin
+}
+
+// Sequential is a policy that selects hosts based on sequential ordering.
+type Sequential struct{}
+
+var _ Policy = &Sequential{}
+
+// String return the name of policy Sequential
```
Comment (miekg) [inline plugin/pkg/policy/policy.go]:
```suggestion
// String returns the name of policy Sequential.
```

**Review by miekg — CHANGES_REQUESTED** [review]

**Comment by miekg** [comment]
yeah, 50 lines reduction in code :)

**Pull request merged**